### DISCUSSION TITLE
How do I remap the key J to Ctrl+Alt+Up arrow for the arc browser

### DISCUSSION THREAD

**Discussion by rp1231** [gdh1995/vimium-c#1138] [Q&A] [answered]
On the Arc browser the next and previous tab shortcuts are Ctrl+Alt +Down and Ctrl+Alt+Up respectively.
How do I map J and K to these shortcuts?
From what I inferred from the documentation, 
Something like this isn't possible:
map J <a-c-\<up\>>
map K <a-c-\<down\>>

**Comment by rp1231** [comment]
@gdh1995 is this possible?
Thanks

**Comment by gdh1995** [comment, marked as answer]
No you can't do so by Vimium C.

Vimum C's keyboard layout rules, `map` and mapKey directives can only take effect inside Vimium C itself.

**Reply by rp1231** [reply]
Would it be possible for you to add code to vimium c that implements these shortcuts instead of the standard ones if arc browser is detected.
Thanks

**Reply by gdh1995** [reply]
Sorry I can't get what you want.

If you just want a key to switch to another tab, then there have been such default key mappings:
```
map J previousTab
map K nextTab
map gT previousTab
map gt nextTab
```

However, the Arc browser has its own private logic of sorting tabs, while extensions can not get any real order. As a result, `nextTab` may jump to one of some unexpected tabs.

**Reply by rp1231** [reply]
Yes, nextTab and previousTab don't work properly in the arc browser......
But the keyboard shortcuts Ctrl + Alt + Up and Ctrl + Alt + Down work fine for navigating to the next and previous tab, which is why I wanted to map them to J and K. 
Hope I'm getting my point across.

**Reply by gdh1995** [reply]
> Vimum C's keyboard layout rules, map and mapKey directives can only take effect inside Vimium C itself.

There's no API to "tell Arc to run Ctrl+Alt+Up".

**Reply by rp1231** [reply]
Oh ok, I see.
Thanks

**Reply by Zacke0** [reply]
I got around this by remapping the next/previous tabs to ctrl+j/k in Arc's settings. This is a bit less comfortable than shift-j/k with a standard MacOS keyboard layout, but I've used Karabiner-Elements to remap the backtick (`) key to register as ctrl when held down. Massive QOL improvement for navigating tabs in Arc.

**Reply by rp1231** [reply]
The problem is that on windows, Arc doesn't support remapping keyboard shortcuts yet.

**Reply by alexwforsythe** [reply]
You can remap keys in Windows with [Microsoft PowerToys](https://apps.microsoft.com/detail/XP89DCGQ3K6VLD?hl=en-US&gl=US&ocid=pdpshare)' Keyboard Manager. Note that remapping <kbd>ctrl</kbd>+<kbd>j</kbd>/<kbd>k</kbd> will disable them for navigating up and down in the Vomnibar tabs menu.
